saleor/product/utils/costs.py

```diff
@@ -21,56 +21,63 @@ def __post_init__(self):
 
 
 def get_product_costs_data(
-    product: "Product",
+    product_channel_listing: "ProductChannelListing",
+    variant_channel_listings: Iterable[ProductVariantChannelListing],
 ) -> Tuple[MoneyRange, Tuple[float, float]]:
 
     purchase_costs_range = MoneyRange(start=zero_money(), stop=zero_money())
     margin = (0.0, 0.0)
 
+    product = product_channel_listing.product
     if not product.variants.exists():
```

Comment:
```suggestion
    if not product_channel_listing.product.variants.exists():
```
![image](https://user-images.githubusercontent.com/43955230/94915546-5e20e700-04ad-11eb-9aa7-6cc22cfcfcfb.png)

Reply:
But we still not able to optimize it through data loader. :(

Reply:
Maby we should add `variant_list` as param?